### PR TITLE
Provide clear selection of text in terminal view

### LINKED ISSUE #591: Pressing ESC should clear selection
Pressing ESC should clear selection as in native `terminal` and `Iterm2`

**Steps to reproduce:**
- Open HyperTerm
- Press `CMD+A` to select everything
- Press `ESC`

**Expected:**
- Selection to cleared

**Actual:**
- Selection persisted.

### PR DESCRIPTION
**Pull request by ppot**
- Click on Terminal view now deselect text.
- Valid keyboard event now deselect text.
- Add configuration option for copy-on-select parameter

Fixes #591

### CODE REVIEW

**Comment by leo** [comment]
@ppot Great fix! Please rebase though.

File: lib/components/term.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ export default class Term extends Component {
     this.term.prefs_.set('receive-encoding', 'raw');
     this.term.prefs_.set('send-encoding', 'raw');
     this.term.prefs_.set('alt-sends-what', 'browser-key');
+    this.term.prefs_.set('copy-on-select', false);
```
Comment (rauchg) [inline lib/components/term.js]: For another PR, this could be an option. A lot of Linux users are used to copy on select from xterm

Comment (ppot) [inline lib/components/term.js]: Got it. I will work on an option setting for this. Any preferences ?

**Comment by ppot** [comment]
@leo Rebased and ready for a merge!

**Comment by ppot** [comment]
@rauchg  Configuration add for copy-on-select to this PR!

File: app/config-default.js [hunk 2]
```diff
@@ -60,7 +60,10 @@ module.exports = {
     env: {},
 
     // set to false for no bell
-    bell: 'SOUND'
+    bell: 'SOUND',
+
+    // set to false for no copy-on-select
```
Comment (rauchg) [inline app/config-default.js]: the comment should be more descriptive? Like "If `true`, selected text will automatically be copied to the clipboard.

File: lib/components/term.js [hunk 3]
```diff
@@ -45,6 +45,12 @@ export default class Term extends Component {
       this.term.prefs_.set('audible-bell-sound', '');
     }
 
+    if (props.copyOnSelect) {
+      this.term.prefs_.set('copy-on-select', true);
+    } else {
+      this.term.prefs_.set('copy-on-select', false);
+    }
+
```
Comment (rauchg) [inline lib/components/term.js]: This should also be in `componentWillReceiveProps`

Comment (ppot) [inline lib/components/term.js]: Updated and rebased in last commit.

**Comment by leo** [comment]
Thanks! ❤️

**Pull request merged**